src/FormFile.js

```diff
@@ -0,0 +1,186 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useContext, useMemo } from 'react';
+import all from 'prop-types-extra/lib/all';
+import Feedback from './Feedback';
+import FormFileInput from './FormFileInput';
+import FormFileLabel from './FormFileLabel';
+import FormContext from './FormContext';
+import { useBootstrapPrefix } from './ThemeProvider';
+
+const propTypes = {
+  /**
+   * @default 'form-file'
+   */
+  bsPrefix: PropTypes.string,
+
+  /**
+   * A seperate bsPrefix used for custom controls
+   *
+   * @default 'custom-file'
+   */
+  bsCustomPrefix: PropTypes.string,
+
+  /**
+   * The FormFile `ref` will be forwarded to the underlying input element,
+   * which means it will be a DOM node, when resolved.
+   *
+   * @type {ReactRef}
+   * @alias ref
+   */
+  _ref: PropTypes.any,
```

Comment:
This component is already forwarding its ref properly, so why do we need this as a part of its API?

Reply:
Probably no reason at all. Left from FormCheck that I used as a basis for this. Removing since refs seems to be working without it.